enhancements/authentication/configuring-webhook-token-authenticators.md

```diff
@@ -0,0 +1,176 @@
+
+Any secret reference which gets set to this field has to be validated. In 4.4, it will only be
+possible to configure a single element for the `wehbhookTokenAuthenticators` field and the referenced
+kubeconfigs will only be allowed to use the `-data` forms of their fields to configure
+certificate authority, client certificate and key.
+
+This path is compatible with kubernetes as it is possible to directly map the referenced
+secrets' contents to files and specify these in the kube-apiserver configuration.
+
+#### Backend - kube-apiserver-operator
+
+- observes the changes in the `authentication/cluster` resource and performs validation of any references
+  in `webhookTokenAuthenticators` field
+- it pushes any webhook authenticator that passes the validation to `openshift-kube-apiserver/webhooktokenauthenticators`
+  secret, where `key=authenticator_name` where `authenticator_name` is either the name of the referenced secret or
+  the CR
+- if no webhook authenticators pass validation, the operator keeps the previous state
+- redeploys the kube-apiserver static pods while specifying the new
+  mounted files in the `webhookTokenAuthenticators` [config field](https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/kubecontrolplane/v1/types.go#L146).
+
+#### Backend - Integrated OAuth Server Configuration
+
+It's cluster-authentication-operator's responsibility to:
+- observe `authentication.config/cluster` field
+- if set to "IntegratedOAuth", it creates secret `openshift-config/integrated-oauth` and adds it to
+  the `webhookTokenAuthenticators` field of the `authentication.config/cluster` resource
+- the above secret contains details that point kube-apiserver to the oauth-apiserver for token reviews
+
+To perform token reviews for tokens issued by the integrated oauth-server, the new oauth-apiserver needs to
+be able to handle token review requests, such capability is currently missing and needs to be implemented.
+
+When responding after a successful authentication, the oauth-apiserver needs to return all of the users
+groups, on top of which it needs to add the `system:authenticated:oauth` group to keep the behavior of
+previous verions. This group adds self-provisioner cluster role to the authenticated user, the role
+allows users privileges necessary to work with projects.
+
+Similarly, if any integrator would like their users to have project privileges, they need to add this
+specific group in their token review responses.
+
+### Risks and Mitigations
+
+There are a couple of issues that root in the very fact of making it possible of
+replacing the oauth-server with a custom token authenticator:
+
+- `oauth-proxy` relies on the _/.well-known/oauth-authorization-server_ endpoint
+  and if the integrated oauth-server is turned off and the `oauthMetadata` is not set
+  properly, things like grafana and prometheus won't be accessible
+- function of selfubjectaccessreviews and the "users/~" echo endpoint may not work
```

Comment:
why would this fail?

Reply:
I think I got this from some old document but since SSAR is pure Kube and "users/~" is just an endpoint being accessed with a token, I think we should be fine now, will remove in next revision.

Reply:
nit: -> selfsubjectaccessreviews